### PR TITLE
WT-9090 Define a new user operation function in the cpp testing

### PR DESCRIPTION
**Pull request by etienneptl**
A new `custom_operation` has been defined in the `database_operation` base class so one can use it as an entry point of any kind of operation in their test.
Note: Testing of this function will be done as part of WT-9095.

### CODE REVIEW

**Comment by luke-pearson** [comment]
I don't like the name "user_operation" as it implies the other operations aren't user defined operations.

**Comment by etienneptl** [comment]
> I don't like the name "user_operation" as it implies the other operations aren't user defined operations.

I renamed it to `custom_operation`, see 99eb272edf5cea086b3d161f432b9f2c70d8b21a and f055c7f4b751bfbd5aa1808b378d5837a801e9e4.

File: dist/test_data.py [hunk 1]
```diff
@@ -164,18 +165,21 @@ def __ge__(self, other):
 # Configuration that applies to the workload_generator component.
 #
 workload_generator = enabled_config_true + component_config + [
+    Config('custom_config', '',r'''
+        Config that specifies the number of user operation threads and their behaviour.''',
```
Comment (luke-pearson) [inline dist/test_data.py:169]: Can we make this a bit more explicit? Something like "Config that specifies the number of custom_operation threads and their behaviour." The name "custom_config" seems a bit vague to me, so if we update the definition it'll help.

File: test/cppsuite/test_harness/util/api_const.h [hunk 2]
```diff
@@ -47,6 +47,7 @@ extern const std::string CACHE_SIZE_MB;
 extern const std::string CC_PAGES_REMOVED;
 extern const std::string COLLECTION_COUNT;
 extern const std::string COMPRESSION_ENABLED;
+extern const std::string CUSTOM_CONFIG;
```
Comment (luke-pearson) [inline test/cppsuite/test_harness/util/api_const.h:50]: I still feel like this is a bit confusing, I know its not consistent but we may want to think of making that config "custom_op_config" or something.

**Comment by luke-pearson** [comment]
LGTM other than naming issue.

**Review by luke-pearson — APPROVED** [review]

File: dist/test_data.py (hunk 1 above)
Comment (etienneptl) [reply dist/test_data.py]: Yep, thanks for the suggestion, see https://github.com/wiredtiger/wiredtiger/pull/7810/commits/6c119649ede04c59234496631b17521fe094ed98

File: test/cppsuite/test_harness/util/api_const.h (hunk 2 above)
Comment (etienneptl) [reply test/cppsuite/test_harness/util/api_const.h]: I have renamed the different constants to make it clearer, see b0d25586d4a017f10a2ab15c2282c3d5c35e6760

**Review by Siddhartha8899 — APPROVED** [review]

**Pull request merged**